drivers/unix/file_access_unix.cpp

```diff
@@ -104,6 +108,26 @@ Error FileAccessUnix::_open(const String &p_path, int p_mode_flags) {
 		}
 	}
 
+#if defined(OSX_ENABLED) && defined(TOOLS_ENABLED)
+	if (p_mode_flags == READ) {
+		DIR *dir = opendir(path.get_base_dir().utf8().get_data());
+		struct dirent *dir_entry;
+		if (dir) {
+			String base_file = path.get_file();
+			while ((dir_entry = readdir(dir)) != NULL) {
```

Comment:
This is very inefficient, please look for a different way to do it. Maybe the stat() function is a better fit for this?

Reply:
The problem is that stat doesn't return the name of the requested file, so there is no way to check the case. I also didn't find a function similar to FindFirstFileW on windows. I'll look into another solution.